### PR TITLE
Mandatory prefix for custom elements

### PR DESCRIPTION
**Pull request by TomasKulhanek**
This PR adds mandatoryPrefix property. By default it is false - thus compatible with previous behavior. 
* `mandatoryPrefix=false`, then `Customelement` is registered as `<au-customelement>`, but `CustomElement` is registered as `<custom-element>`. 
* `mandatoryPrefix=true`, then `Customelement` is registered with fallbackPrefix as `<au-customelement>`, and `CustomElement` is registered with fallback prefix too: `<au-custom-element>`.

### CODE REVIEW

**Comment by bigopon** [comment]
@TomasKulhanek thanks for the PR, could you help revert all the dist files? they will be built and added during release process

**Comment by bigopon** [comment]
This seems to be a useful feature 👍 to help release the burden of naming. Naming is hard though, and `mandatoryPrefix` feels difficult to reason about, as it doesn't tell the mean "a prefix will always be added regardless CE name". Maybe use `alwaysPrefix`?

File: src/custom-element-registry.ts [hunk 1]
```diff
@@ -67,7 +69,7 @@ export class CustomElementRegistry implements ICustomHtmlRegistry {
       classDefinition: classDefinition
     };
 
-    if (tagName.indexOf('-') === -1) {
+    if (this.mandatoryPrefix || tagName.indexOf('-') === -1) {
```
Comment (bigopon) [inline src/custom-element-registry.ts:72]: what if the element name already starts with the prefixed, do we check that, or do we prefix it again? Example:
- does `bdl-range` become `bdl-bdl-range`?

File: README.md [hunk 2]
```diff
@@ -147,6 +152,6 @@ To run the unit tests, first ensure that you have followed the steps above in or
 ## How it works
 
 * Each of custom element will be backed by a view model.
-* For each view model class, a corresponding native custom element class will be created and defined, with the name derived from metadata and fallbacks to view model class name. If there is no hyphen `-` in the name of a custom element view model, a prefix (`au-` by default) will be added to the name. This can be change in `CustomElementRegistry` instance.
+* For each view model class, a corresponding native custom element class will be created and defined, with the name derived from metadata and fallbacks to view model class name. If there is no hyphen `-` in the name of a custom element view model, a prefix (`au-` by default) will be added to the name. This can be changed in `CustomElementRegistry` instance.
```
Comment (bigopon) [inline README.md:155]: Nice, thanks!

File: src/custom-element-registry.ts (hunk 1 above)
Comment (TomasKulhanek) [reply src/custom-element-registry.ts:72]: - Good point, currently it is prefixed again. I can consider whether the prefix is already there so compare fallbackprefix with tagname.startsWith. 
- regarding name `mandatoryPrefix`, `alwaysPrefix` sounds good as well, or what about `forcePrefix` - it is boolean, so might be meaningful.

Comment (bigopon) [reply src/custom-element-registry.ts:72]: I was considering between `always` vs `force`, and `force` will like it's leaning towards more error checking that signaling the effect of this flag. Probably we should consider:
- no ambiguity (force: erring or helping?)
- short (force: shorter but similar length with always)

Comment (TomasKulhanek) [reply src/custom-element-registry.ts]: * I put forcePrefix - shorter
* Added check to avoid double prefix au-au-component or custom-custom-element

Comment (bigopon) [reply src/custom-element-registry.ts]: Well, I like your stance, I'm not 100% on `always` so let's go with `force`

**Comment by bigopon** [comment]
Ok, maybe let's leave the naming thing to later. Thanks for the PR. Let's go

**Pull request merged**